### PR TITLE
sanitize strings for use as keys

### PR DESCRIPTION
**Pull request by klevron1**

### CODE REVIEW

File: consul_awx.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def add_to_group(self, group, host, parent=None):
         # https://docs.ansible.com/ansible/latest/network/getting_started/first_inventory.html
```
Comment (wilfriedroset) [inline consul_awx.py]: please move the documentation to the method definition

File: consul_awx.py [hunk 2]
```diff
@@ -123,6 +124,11 @@ def get_node_services(self, node):
         return self.get_node(node)["Services"]
 
 
+def sanitize(string):
+    sanitized = re.sub(r"[^A-Za-z0-9]", "_", string)
```
Comment (wilfriedroset) [inline consul_awx.py]: could be a oneliner

**Comment by wilfriedroset** [comment]
I've added a couple of suggestion (cosmetics).
Could you update the unittest to make sure we do not introduce a regression later on.

For example here https://github.com/wilfriedroset/consul-awx/blob/master/consul_awx_test.py#L60 you can update the service name to include a dash in the name.

File: consul_awx.py (hunk 1 above)
Comment (klevron1) [reply consul_awx.py]: Updated

File: consul_awx.py (hunk 2 above)
Comment (klevron1) [reply consul_awx.py]: updated

**Comment by wilfriedroset** [comment]
Could you also update the unit test?

**Comment by klevron1** [comment]
Added a test case for the service sanitization.
Without the `service = sanitize(service)` line, the test fails as follows:
```
E                   KeyError: 'd-d'

consul_awx.py:90: KeyError
```

**Review by wilfriedroset — APPROVED** [review]

**Pull request merged**